### PR TITLE
Implement read/write reserved state and query_commit_path in RawRepository

### PR DESCRIPTION
**Pull request by JeongHunP**

### CODE REVIEW

**Comment by TomTaehoonKim** [comment]
sorry for the miscommunication.

can you update according to #196 please?

File: repository/src/raw/implementation.rs [hunk 1]
```diff
@@ -362,10 +362,48 @@ impl RawRepositoryImplInner {
...
+        if merge_base != ancestor {
+            return Err(Error::InvalidRepository(
+                "ancestor is not the merge base of two commits".to_string(),
+            ));
+        }
+
+        let descendant_oid = Oid::from_bytes(&descendant.hash)?;
+        let ancestor_oid = Oid::from_bytes(&ancestor.hash)?;
+        //let ancestor_commit = self.repo.find_commit(ancestor_oid)?;
+        //let ancestor_parent_oid = ancestor_commit.parent(0)?.id();
```
[the remark this answers is not in the document]
Comment (TomTaehoonKim) [reply repository/src/raw/implementation.rs:384]: probably due to my request of applying #196, but should've deleted

[the remark this answers is not in the document]
Comment (JeongHunP) [reply repository/src/raw/implementation.rs:384]: Yes, it should be deleted.

[the remark this answers is not in the document]
Comment (JeongHunP) [reply repository/src/raw/implementation.rs:384]: fixed

**Review by junha1 — APPROVED** [review]

**Pull request merged**